lib/plugins/physics.js

```diff
@@ -223,44 +223,44 @@ function inject (bot) {
-    push.y = push.y/physics.gravity;
+    const flow = liquidFlow(pos)
+    const push = flow.clone().scaled(physics.liquidPush)
+    push.y = push.y / physics.gravity
     return push
   }
 
   function liquidFlow (pos) {
-    const flow = new Vec3(0,0,0);
+    const flow = new Vec3(0, 0, 0)
     const block = bot.blockAt(pos)
-    const base = block.metadata;
-    const id = block.type;
-    if(base > 7) {
-      base = 0;
-      flow.set(0,-1,0);
+    const base = block.metadata
+    const id = block.type
+    if (base > 7) {
+      base = 0
+      flow.set(0, -1, 0)
     }
     const depth = level => {
-      if(level > 7) return 0;
-      return level; 
+      if (level > 7) return 0
+      return level
     }
     const Blocks = {
-      'x+': bot.blockAt(pos.offset(1,0,0)),
-      'x-': bot.blockAt(pos.offset(-1,0,0)),
-      'z+': bot.blockAt(pos.offset(0,0,1)),
-      'z-': bot.blockAt(pos.offset(0,0,-1))
+      'x+': bot.blockAt(pos.offset(1, 0, 0)),
+      'x-': bot.blockAt(pos.offset(-1, 0, 0)),
+      'z+': bot.blockAt(pos.offset(0, 0, 1)),
+      'z-': bot.blockAt(pos.offset(0, 0, -1))
     }
     const waterBlocks = {}
-    for(block of Object.entries(Blocks)){
-      if(Blocks[key].type == id) waterBlocks[key] = depth(Blocks[key].metadata)
+    for (block of Object.entries(Blocks)) {
+      if (Blocks[key].type == id) waterBlocks[key] = depth(Blocks[key].metadata)
     }
-    if(waterBlocks['x+'] !== null && waterBlocks['x+'] > base) flow = flow.add(1,0,0);
-    if(waterBlocks['x-'] !== null && waterBlocks['x-'] > base) flow = flow.add(-1,0,0);
-    if(waterBlocks['z+'] !== null && waterBlocks['z+'] > base) flow = flow.add(0,0,1);
-    if(waterBlocks['z-'] !== null && waterBlocks['z-'] > base) flow = flow.add(0,0,-1);
-    if(waterBlocks['x+'] !== null && waterBlocks['x+'] < base && flow.x == 0) flow = flow.add(-1,0,0)
-    if(waterBlocks['x-'] !== null && waterBlocks['x-'] < base && flow.x == 0) flow = flow.add(1,0,0);
-    if(waterBlocks['z+'] !== null && waterBlocks['z+'] < base && flow.z == 0) flow = flow.add(0,0,-1)
-    if(waterBlocks['z-'] !== null && waterBlocks['z-'] < base && flow.z == 0) flow = flow.add(0,0,1);
-    return flow;
+    if (waterBlocks['x+'] !== null && waterBlocks['x+'] > base) flow = flow.add(1, 0, 0)
+    if (waterBlocks['x-'] !== null && waterBlocks['x-'] > base) flow = flow.add(-1, 0, 0)
+    if (waterBlocks['z+'] !== null && waterBlocks['z+'] > base) flow = flow.add(0, 0, 1)
+    if (waterBlocks['z-'] !== null && waterBlocks['z-'] > base) flow = flow.add(0, 0, -1)
+    if (waterBlocks['x+'] !== null && waterBlocks['x+'] < base && flow.x == 0) flow = flow.add(-1, 0, 0)
+    if (waterBlocks['x-'] !== null && waterBlocks['x-'] < base && flow.x == 0) flow = flow.add(1, 0, 0)
+    if (waterBlocks['z+'] !== null && waterBlocks['z+'] < base && flow.z == 0) flow = flow.add(0, 0, -1)
+    if (waterBlocks['z-'] !== null && waterBlocks['z-'] < base && flow.z == 0) flow = flow.add(0, 0, 1)
```

Comment:
still trying to update a constant...
Also, flow.x/flow.z should be compared with strict type comparsion (===)

Reply:
.add updates current value. Don't reassign it.

Reply:
Still, make a map how flow should change at each neighbor and apply it dynamically in the `for` above

Reply:
flow.add(0, 0, 1) won't work, add takes a Vec3 as a parameter, needs to be flow.add(new Vec3(0, 0, 1))